addons/im_livechat/controllers/chatbot.py

```diff
@@ -101,14 +101,15 @@ def chatbot_validate_email(self, channel_id):
 
         # sudo: chatbot.script - visitor can access chatbot script of their channel
         chatbot = discuss_channel.sudo().chatbot_current_step_id.chatbot_script_id
-        user_messages = discuss_channel.message_ids.filtered(
-            lambda message: message.author_id != chatbot.operator_partner_id
-        )
-
-        if user_messages:
-            user_answer = user_messages.sorted(lambda message: message.id)[-1]
-            result = chatbot._validate_email(user_answer.body, discuss_channel)
-
+        domain = [
+            ("author_id", "!=", chatbot.operator_partner_id.id),
+            ("model", "=", "discuss.channel"),
+            ("res_id", "=", channel_id),
+        ]
+        last_user_message = self.env["mail.message"].sudo().search(domain, order="id desc", limit=1)
```

Comment:
used twice, might be worth adding a dedicated helper? or even computed field?

Reply:
The "index" is a tuple (specific author in specific channel) so it can't be a compute on one of the models.

Not sure if worth having a method for only 2, it's just a simple search at the end. Maybe if we have more or if we need to do batch computing some day.